vic/drivers/classic/src/parse_output_info.c

```diff
@@ -59,96 +58,106 @@ parse_output_info(FILE                  *gp,
-                            "specified before you can specify \"OUTFILE\".");
-                }
-                if (outfilenum >= (short int)options.Noutfiles) {
-                    log_err("number of output files specified in N_OUTFILES "
-                            "(%zu) is less than actual number of output files "
-                            "defined in the global param file.",
-                            options.Noutfiles);
-                }
-                sscanf(cmdstr, "%*s %s", (*out_data_files)[outfilenum].prefix);
+    // Count the number of output files listed in the global param file
+    options.Noutfiles = count_n_outfiles(gp);
 
-                // determine how many variable will be in this file before
-                // allocating (GH: 209)
-                (*out_data_files)[outfilenum].nvars = count_outfile_nvars(gp);
+    // only parse the output info if there are output files to parse
+    if (options.Noutfiles > 0) {
+        free_out_data_files(out_data_files);
 
-                (*out_data_files)[outfilenum].varid =
-                    calloc((*out_data_files)[outfilenum].nvars,
-                           sizeof(*((*out_data_files)[outfilenum].varid)));
-                outvarnum = 0;
-            }
-            else if (strcasecmp("OUTVAR", optstr) == 0) {
-                if (outfilenum < 0) {
-                    log_err("Error in global param file: \"OUTFILE\" must be "
-                            "specified before you can specify \"OUTVAR\".");
-                }
-                strcpy(format, "");
-                strcpy(typestr, "");
-                strcpy(multstr, "");
-                sscanf(cmdstr, "%*s %s %s %s %s", varname, format, typestr,
-                       multstr);
-                if (strcasecmp("", format) == 0) {
-                    strcpy(format, "*");
-                    type = OUT_TYPE_DEFAULT;
-                    mult = 0; // 0 means default multiplier
-                }
-                else {
-                    if (strcasecmp("OUT_TYPE_USINT", typestr) == 0) {
-                        type = OUT_TYPE_USINT;
-                    }
-                    else if (strcasecmp("OUT_TYPE_SINT", typestr) == 0) {
-                        type = OUT_TYPE_SINT;
+        *out_data_files = calloc(options.Noutfiles,
+                                 sizeof(*(*out_data_files)));
+        init_output_list(out_data, false, "%.4f", OUT_TYPE_FLOAT, 1);
+
+        // PRT_SNOW_BAND is ignored if N_OUTFILES has been specified
+        options.PRT_SNOW_BAND = false;
+
+        while (!feof(gp)) {
+            if (cmdstr[0] != '#' && cmdstr[0] != '\n' && cmdstr[0] != '\0') {
+                sscanf(cmdstr, "%s", optstr);
+
+                if (strcasecmp("OUTFILE", optstr) == 0) {
+                    outfilenum++;
+                    if (!options.Noutfiles) {
+                        log_err("in global param file: \"N_OUTFILES\" must be "
```

Comment:
actually, this check should go altogether.  We already check that `Noutfiles > 0` in line 66 so we're never going to get here.

I've also done a grep search to make sure we aren't missing any other mentions of `Noutfiles`.